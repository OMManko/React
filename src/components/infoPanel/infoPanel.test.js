import { mount } from 'enzyme';
import React from 'react';
import InfoPanel from '../infoPanel/infoPanel';
import MovieInfo from "../movieInfo/movieInfo";
import SearchForm from "../searchForm/searchForm";

let component;

const movieInfoMock = {
    src: "https://cdn.cnn.com/cnnnext/dam/assets/190315113313-04-mafia-crime-families-have-evolved-file-restricted-use-super-tease.jpg",
    title: "Pulp Fiction",
    year: 1992,
    genre: "Oscar-winning Movies",
    duration: "154 min",
    description: "Some descrition"
};

describe('<InfoPanel />', () => {
    beforeEach(() => {
        component = mount(
            <InfoPanel />
        );
    });

    afterEach(() => {
        component.unmount();
    });

    it('should render info panel and match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should contain movie info', () => {
        expect(component).toMatchSnapshot();
        expect(component.contains(<MovieInfo movie={movieInfoMock}/>)).toBe(true);
    });

    it('should contain search form', () => {
        expect(component).toMatchSnapshot();
        expect(component.contains(<SearchForm/>)).toBe(true);
    });
});
