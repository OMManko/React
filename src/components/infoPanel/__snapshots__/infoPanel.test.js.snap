// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoPanel /> should contain movie info 1`] = `ReactWrapper {}`;

exports[`<InfoPanel /> should contain search form 1`] = `ReactWrapper {}`;

exports[`<InfoPanel /> should render info panel and match snapshot 1`] = `ReactWrapper {}`;
